enhancements/sig-architecture/33-detached-deploy-mode/detached-deploy-mode.md

```diff
@@ -0,0 +1,224 @@
+# Running deployments of cluster-manager and klusterlet outside of Hub and ManagedCluster
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [website](https://github.com/open-cluster-management/website/)
+
+## Summary
+
+The proposed work would support running deployments of cluster-manager and klusterlet outside of Hub and ManagedCluster. 
+
+## Motivation
+
+Currently, the cluster manager and klusterlet should be deployed on the hub/managed cluster, it is mandatory to expose the hub cluster to the managed cluster even if the managed cluster resides in an insecure environment. 
+
+To address this, we propose to provide an option for running cluster-manager/klusterlet outside of the hub/managed cluster.
+
+This ability will bring some additional possible benefits:
+* We can leverage compute resources better by running several cluster-managers/klusterlets deployments on the same powerful cluster; it also enhances our management of all deployments.
+* The footprints of the hub/managed cluster can be reduced.
+* We can use this ability to run the klusterlet on the hub cluster, so we do not need to expose the hub cluster to the managed cluster, and accordingly, we expose the managed cluster to the hub (A bit like Pull Mode to Push Mode).
+
+### Goals
+
+* Support deploy cluster-manager outside of the hub cluster.
+* Support deploy klusterlet outside of the managed cluster.
+
+### Non-Goals
+
+* Support deploy addons agents outside of the managed cluster.(TBD)
+* Authenticate managed cluster without using CSR(CertificateSigningRequest).
+
+## User Stories
+
+### Story1
+
+As a managed cluster administrator, I hope as few as possible workloads running in my cluster while the cluster can be managed by the hub.
+
+### Story2
+
+As a hub cluster administrator, on the premise that an untrusted cluster can be managed, I do not want the cluster to talk to the hub cluster directly.
```

Comment:
This seems to be a call for a push model

Reply:
Yes, that's right. It could be solved by a push model as well.